src/main/groovy/spock/genesis/transform/GenASTTransformation.groovy

```diff
@@ -0,0 +1,144 @@
+package spock.genesis.transform
+
+import groovy.transform.CompileStatic
+import org.codehaus.groovy.ast.ASTNode
+import org.codehaus.groovy.ast.AnnotatedNode
+import org.codehaus.groovy.ast.AnnotationNode
+import org.codehaus.groovy.ast.ClassHelper
+import org.codehaus.groovy.ast.ClassNode
+import org.codehaus.groovy.ast.MethodNode
+import org.codehaus.groovy.ast.ModuleNode
+import org.codehaus.groovy.ast.expr.ArgumentListExpression
+import org.codehaus.groovy.ast.expr.ConstantExpression
+import org.codehaus.groovy.ast.expr.ConstructorCallExpression
+import org.codehaus.groovy.ast.expr.Expression
+import org.codehaus.groovy.ast.stmt.BlockStatement
+import org.codehaus.groovy.ast.stmt.ReturnStatement
+import org.codehaus.groovy.control.CompilePhase
+import org.codehaus.groovy.control.SourceUnit
+import org.codehaus.groovy.transform.ASTTransformation
+import org.codehaus.groovy.transform.GroovyASTTransformation
+import org.spockframework.compiler.AstNodeCache
+import spock.genesis.generators.LimitedGenerator
+
+/**
+ * Global AST transformation to modify data providers after Spock compilation
+ */
+@CompileStatic
+@GroovyASTTransformation(phase = CompilePhase.CANONICALIZATION)
+@SuppressWarnings(['Instanceof', 'CrapMetric'])
+class GenASTTransformation implements ASTTransformation {
+
+    final static AstNodeCache NODE_CACHE = new AstNodeCache()
+    final static ClassNode ITERATIONS_ANNOTATION = ClassHelper.makeWithoutCaching(Iterations)
+    final static ClassNode LIMITED_CLASS_GENERATOR = ClassHelper.makeWithoutCaching(LimitedGenerator)
+
+    @Override
+    void visit(ASTNode[] nodes, SourceUnit sourceUnit) {
+        ModuleNode module = (ModuleNode) nodes[0]
+        List<ClassNode> classes = module.classes
+
+        for (ClassNode clazz in classes) {
+            if (isSpec(clazz)) {
+                def classIterationsAnnotation = findIterationsAnnotation(clazz)
+                Integer classIterations
```

Comment:
I don't have a test that shows it yet but Iterations can be on the class also and then overridden for each feature as needed.